In [1]:
import pandas as pd
import matplotlib.pyplot as plt
from pandasql import sqldf
pysqldf = lambda q: sqldf(q, globals())

import import_ipynb
from Week1_data_analysis_season import summer_events, winter_events

importing Jupyter notebook from Week1_data_analysis_season.ipynb


### Beyond Descriptive  Statistical

In [2]:
#Summer Olympics:
summer_medal_count = pysqldf('''
                         SELECT
                             Year,
                             COUNT(*) AS total_count,
                             SUM(CASE
                                   WHEN Medal IS NOT NULL THEN 1 ELSE 0
                                 END) AS medal_count,
                             SUM(CASE
                                   WHEN Medal = "Gold" THEN 1 ELSE 0
                                 END) AS gold_count,
                             SUM(CASE
                                   WHEN Medal = "Silver" THEN 1 ELSE 0
                                 END) AS silver_count,
                             SUM(CASE
                                   WHEN Medal = "Bronze" THEN 1 ELSE 0
                                 END) AS bronze_count
                             FROM
                               summer_events
                             GROUP BY
                               Year 
                                   ''')

In [3]:
#Winter Olympics:
winter_medal_count = pysqldf('''
                         SELECT
                             Year,
                             COUNT(*) AS total_count,
                             SUM(CASE
                                   WHEN Medal IS NOT NULL THEN 1 ELSE 0
                                 END) AS medal_count,
                             SUM(CASE
                                   WHEN Medal = "Gold" THEN 1 ELSE 0
                                 END) AS gold_count,
                             SUM(CASE
                                   WHEN Medal = "Silver" THEN 1 ELSE 0
                                 END) AS silver_count,
                             SUM(CASE
                                   WHEN Medal = "Bronze" THEN 1 ELSE 0
                                 END) AS bronze_count
                             FROM
                               winter_events
                             GROUP BY
                               Year 
                                   ''')

**I have generated two tables that tally the overall count of medals awarded in the winter and summer Olympics, respectively. My next step is to compute the Pearson correlation coefficient between these two sets of data.**

In [4]:
summer_medal_count.head()

Year  total_count  medal_count  gold_count  silver_count  bronze_count
0  1896          380          143          62            43            38
1  1900         1936          604         201           228           175
2  1904         1301          486         173           163           150
3  1906         1733          458         157           156           145
4  1908         3101          831         294           281           256

In [5]:
winter_medal_count.head()

Year  total_count  medal_count  gold_count  silver_count  bronze_count
0  1924          460          130          55            38            37
1  1928          582           89          30            28            31
2  1932          352           92          32            32            28
3  1936          895          108          36            37            35
4  1948         1075          135          41            48            46

In [6]:
summer_medal_count

Year  total_count  medal_count  gold_count  silver_count  bronze_count
0   1896          380          143          62            43            38
1   1900         1936          604         201           228           175
2   1904         1301          486         173           163           150
3   1906         1733          458         157           156           145
4   1908         3101          831         294           281           256
5   1912         4040          941         326           315           300
6   1920         4292         1308         493           448           367
7   1924         5233          832         277           281           274
8   1928         4992          734         245           239           250
9   1932         2969          647         229           214           204
10  1936         6506          917         312           310           295
11  1948         6405          852         289           284           279
12  1952         8270          897         306           291           300
13  1956         5127          893         302           293           298
14  1960         8119          911         309           294           308
15  1964         7702         1029         347           339           343
16  1968         8588         1057         359           340           358
17  1972        10304         1215         404           392           419
18  1976         8641         1320         438           434           448
19  1980         7191         1384         457           458           469
20  1984         9454         1476         497           477           502
21  1988        12037         1582         520           513           549
22  1992        12977         1712         559           549           604
23  1996        13780         1842         608           605           629
24  2000        13821         2004         663           661           680
25  2004        13443         2001         664           660           677
26  2008        13602         2048         671           667           710
27  2012        12920         1941         632           630           679
28  2016        13688         2023         665           655           703

The Summer Olympics began in 1896 and the Winter Olympics started in 1924, the number of medal counts in their respective arrays are different. As a result, I need to generate a truncated table for the Summer Olympics that only includes data from 1924 onwards to match the length of the Winter Olympics array.

In [7]:
# I exclude the first 7 rows (which correspond to the years before 1924)
summer_medal_count_new = summer_medal_count[7:]

In [8]:
summer_medal_count_new

Year  total_count  medal_count  gold_count  silver_count  bronze_count
7   1924         5233          832         277           281           274
8   1928         4992          734         245           239           250
9   1932         2969          647         229           214           204
10  1936         6506          917         312           310           295
11  1948         6405          852         289           284           279
12  1952         8270          897         306           291           300
13  1956         5127          893         302           293           298
14  1960         8119          911         309           294           308
15  1964         7702         1029         347           339           343
16  1968         8588         1057         359           340           358
17  1972        10304         1215         404           392           419
18  1976         8641         1320         438           434           448
19  1980         7191         1384         457           458           469
20  1984         9454         1476         497           477           502
21  1988        12037         1582         520           513           549
22  1992        12977         1712         559           549           604
23  1996        13780         1842         608           605           629
24  2000        13821         2004         663           661           680
25  2004        13443         2001         664           660           677
26  2008        13602         2048         671           667           710
27  2012        12920         1941         632           630           679
28  2016        13688         2023         665           655           703

Next, I compute the Pearson correlation coefficient for the overall medal counts of the Winter and Summer Olympics between the years 1924 and 2016.

In [9]:
import numpy as np

In [10]:
x_simple = winter_medal_count.medal_count
y_simple = summer_medal_count_new.medal_count
coffi = np.corrcoef(x_simple, y_simple)

In [11]:
coffi

array([[1.        , 0.94141801],
       [0.94141801, 1.        ]])

The Pearson correlation coefficient between the overall medal counts of the Winter and Summer Olympics from 1924 to 2016 is 0.94, indicating a strong positive correlation. This suggests that a country's performance in the Winter Olympics is highly linked to its performance in the Summer Olympics.

**Next, I will compute the standard deviation in country performance over time. To assess the difference, I will compare the mean standard deviation of the Winter Olympics to that of the Summer Olympics.**

In [12]:
std_medal_count_summer_olympics = np.std(y_simple)
std_medal_count_winter_olympics = np.std(x_simple)

In [13]:
print("std_medal_count_summer_olympics =",std_medal_count_summer_olympics)
print("std_medal_count_winter_olympics =",std_medal_count_winter_olympics)

std_medal_count_summer_olympics = 475.323015441357
std_medal_count_winter_olympics = 152.56899942903493


Between 1924 and 2016, the standard deviation in medal count for the Summer Olympics was approximately three times greater than that of the Winter Olympics. This indicates that a country's performance varies more from year to year in the Summer Olympics.